tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/DynamoDb/AwsDynamoDbCommon.cs

```diff
@@ -87,5 +88,25 @@ public static void TagBatchRequest<TBatchRequest>(TBatchRequest request, AwsDyna
                 TagTableNameAndResourceName(tableName, tags, scope);
             }
         }
+
+        public static string GetValueFromDynamoDbAttribute(IDynamoDbAttributeValue value)
+        {
+            if (value.S != null)
+            {
+                return value.S;
+            }
+
+            if (value.N != null)
+            {
+                return value.N;
+            }
+
+            if (value.B != null)
+            {
+                return Encoding.UTF8.GetString(value.B.ToArray());
```

Comment:
`MemoryStream.ToArray()` always allocates a new byte array which we can try to avoid if possible. Something like this could work since there's a `Encoding.UTF8.GetString(Span<byte>)` overload in newer versions of .NET:

```csharp
if (value.B != null)
{
#if NETCOREAPP3_1_OR_GREATER
    if (value.B.TryGetBuffer(out var buffer))
    {
        return Encoding.UTF8.GetString(buffer.AsSpan());
    }
#endif

    // fallback always copies bytes into a new array
    return Encoding.UTF8.GetString(value.B.ToArray());
}
```

(We can probably do better for the older runtimes, too, but I don't think it's worth the extra complexity of dealing with pointers.)